src/services/infra/StatsService.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable import/prefer-default-export */
+import StatsDClient, { StatsD } from "hot-shots"
+import _ from "lodash"
+import { ModelStatic } from "sequelize"
+
+import { config } from "@config/config"
+
+import {
+  GH_MAX_REPO_COUNT,
+  GITHUB_ORG_REPOS_ENDPOINT,
+  ISOMERPAGES_REPO_PAGE_COUNT,
+} from "@root/constants"
+import { AccessToken, Site, User } from "@root/database/models"
+
+import { genericGitHubAxiosInstance } from "../api/AxiosInstance"
+
+export class StatsService {
+  private readonly statsD: StatsD
+
+  private readonly usersRepo: ModelStatic<User>
+
+  private readonly accessTokenRepo: ModelStatic<AccessToken>
+
+  private readonly sitesRepo: ModelStatic<Site>
+
+  constructor(
+    statsDClient: StatsD,
+    usersRepo: ModelStatic<User>,
+    accessTokenRepo: ModelStatic<AccessToken>,
+    sitesRepo: ModelStatic<Site>
+  ) {
+    this.statsD = statsDClient
+    this.usersRepo = usersRepo
+    this.accessTokenRepo = accessTokenRepo
+    this.sitesRepo = sitesRepo
+  }
+
+  submitV1Count = (path: string) => {
+    this.statsD.increment("versions.v1", {
+      version: "v1",
+      path,
+    })
+  }
+
+  countDbUsers = async () => {
+    // NOTE: Track only active users.
+    const numUsers = await this.usersRepo.count({
+      where: {
+        deletedAt: null,
+      },
+    })
+
+    // NOTE: Technically our e2e user will be here also
+    // but over-counting by 1 here is acceptable.
+    this.statsD.distribution("users.email.current", numUsers, 1, {
+      version: "v2",
+    })
+  }
+
+  countGithubSites = async () => {
+    const accessToken = await this.accessTokenRepo.findOne()
+    // NOTE: Cannot submit metrics if we are unable to get said metric
+    if (!accessToken) return
+
+    const sitesArr = await Promise.all(
+      _.fill(Array(ISOMERPAGES_REPO_PAGE_COUNT), null)
+        .map((__, idx) => ({
+          per_page: GH_MAX_REPO_COUNT,
+          sort: "full_name",
+          page: idx + 1,
+        }))
+        .map((params) =>
+          genericGitHubAxiosInstance
+            .get<unknown[]>(GITHUB_ORG_REPOS_ENDPOINT, {
+              params,
+            })
+            .then(({ data }) => data)
+        )
+    )
+
+    this.statsD.distribution("sites.github.all", sitesArr.flat().length, 1)
+  }
+
+  countMigratedSites = async () => {
```

Comment:
you're entirely right here! datadog wants to collect a stream of ephemeral user interactions and from there, construct an evolving view of the application state. in this case, our metrics are more a snapshot of the application state rather than user interaction. 

this is because product wants these metrics to justify impact and prior to rolling out this PR, i've checked w/ other engineers who've implemented datadog that actually grafana is a better way of collecting such derived data (it's basically a db materialised view).

because we don't have grafana set up atm, the middle-ground for this (between product/eng) is to use dd for now to collect and display such data!

Reply:
sounds good (am assuming that we are not taking any noticeable perf hit), should we document this as a todo/jira ticket then?

Reply:
sure. filed as IS-96